### PR TITLE
fix: recommender system

### LINKED ISSUE #417: Fix the recommender
@agpul 
- [x] recommendations response (from our facade) is extended with the boolean flag `isRandom`
- [x] if any of the ids from the recommender can't be found, return all well-suited objects for user needs
- [x] fix recommendations facade based on changes in the external recommender endpoint request body/response
- [x] remove turn on the random flag from env
- [x] add a message parallel to the random flag to inform about the error from the recommender side, when random generated

@danielkryska 
- [x] add `services` recommendations request
- [x] add `data-sources` recommendations request

### PR DESCRIPTION
**Pull request by danielkryska**
closes #417

### CODE REVIEW

**Review by tomaszandel — APPROVED** [review]

File: backend/app/routes/web/recommendation.py [hunk 1]
```diff
@@ -76,47 +72,45 @@ def __str__(self):
...
-            "page_id": page_id,
-            "panel_id": panels,
+            "page_id": "/search/" + panel_id,
+            "panel_id": _get_panel(panel_id),
             "candidates": [],
             "search_data": {},
         }
 
+        if session:
+            request_body["aai_id"] = session.aai_id
```
Comment (michal-szostak) [inline backend/app/routes/web/recommendation.py:112]: In user actions as `aai_uid` `session.aai_state` is being sent. Is this correct? @Michal-Kolomanski ?

Comment (Michal-Kolomanski) [reply backend/app/routes/web/recommendation.py:112]: Yes, it is. But what is the exact difference between `session.aai_state `and `session.aai_id`?

Comment (michal-szostak) [reply backend/app/routes/web/recommendation.py:114]: `state` zwracany jest przez query param z checkina, `aai_id` jest brany z `aai_id = aai_response["userinfo"]["sub"]`

```
async def auth_checkin(code: str, state: str):
    if not state:
        return RedirectResponse(status_code=400, url=UI_BASE_URL)

    try:
        aai_response = rp_handler.finalize(OIDC_ISSUER, dict(code=code, state=state))

        session_id = uuid4()
        username = aai_response["userinfo"]["name"]
        aai_id = aai_response["userinfo"]["sub"]

        session_data = SessionData(
            username=username,
            aai_state=state,
            aai_id=aai_id,
            session_uuid=str(uuid.uuid4()),
        )
```

**Review by michal-szostak — APPROVED** [review]

File: backend/app/routes/web/recommendation.py (hunk 1 above)
Comment (Michal-Kolomanski) [reply backend/app/routes/web/recommendation.py:114]: We should pass here the same AAI_UID which is associated with the user in the MP's DB. This ID is used to uniquely identify the user across all services. That being said, that has remained me that there is a bug causing duplication of users with the same AAI_UID. We should fix it in the new sprint.

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**